### PR TITLE
Start booting up the new binding system for window

### PR DESCRIPTION
**Pull request by abarth**
Currently this code isn't completely wired up, but this patch is an incremental
step in that direction.

### CODE REVIEW

File: sky/engine/core/dart/window.dart [hunk 1]
```diff
@@ -5,11 +5,11 @@
 part of dart_ui;
 
 typedef void _VoidCallback();
-typedef Scene _FrameCallback(Duration duration);
+typedef void _FrameCallback(Duration duration);
 typedef void _EventCallback(Event event);
 
 class Padding {
```
Comment (Hixie) [inline sky/engine/core/dart/window.dart]: ViewPadding

Comment (Hixie) [inline sky/engine/core/dart/window.dart]: or WindowPadding

File: sky/engine/core/script/dom_dart_state.h [hunk 2]
```diff
@@ -5,20 +5,24 @@
 #ifndef SKY_ENGINE_CORE_SCRIPT_DOM_DART_STATE_H_
```
Comment (Hixie) [inline sky/engine/core/script/dom_dart_state.h]: "DOM" dart state?

**Comment by Hixie** [comment]
LGTM

**Comment by abarth** [comment]
Continuing in https://github.com/flutter/engine/pull/1845

**Pull request closed without merging**